### PR TITLE
Add SOVERSION for Encoder and Decoder shared libraries

### LINKED ISSUE #363: Could you please add a SOVERSION to libraries?
Currently libraries are unversioned which prevents inclusion in Fedora.

### PR DESCRIPTION
**Pull request by eclipseo**
Fixes #363

Signed-off-by: Robert-André Mauchin <zebob.m@gmail.com>

### CODE REVIEW

File: Source/Lib/Encoder/Codec/CMakeLists.txt [hunk 1]
```diff
@@ -1,9 +1,15 @@
-# 
+#
 # Copyright(c) 2019 Intel Corporation
 # SPDX - License - Identifier: BSD - 2 - Clause - Patent
-# 
+#
 # EncoderLib Directory CMakeLists.txt
 
+# Shared Encoder Version
+set(ENC_VERSION_MAJOR 1)
```
Comment (jbeich) [inline Source/Lib/Encoder/Codec/CMakeLists.txt]: According to [semantic versioning](https://semver.org/) until API (and ABI) stabilizes SOVERSION can only be 0. For one, 653d26c0f32e moved `stat_report` in the struct -> old consumers may assign value to the wrong field.

Comment (hassount) [reply Source/Lib/Encoder/Codec/CMakeLists.txt]: agreed, code is not yet stable for a 1.x similarly with the decoder

**Comment by eclipseo** [comment]
Rebased.
Used version 0.5.0 following: https://github.com/OpenVisualCloud/SVT-AV1/pull/396

**Comment by 1480c1** [comment]
@hassount, are the encoder and decoder going to be considered to have separate versions? As in once the encoder hits 1.0.0 status, will the decoder be considered 1.0.0? Or will the development be considered separate?

**Comment by hassount** [comment]
the process will probably be different for both so I think we can use different versions for now and Synch after we hit a 1.0 that said, I would let @joelsole @andrey-norkin @kylophone  jump in here

**Comment by kylophone** [comment]
LGTM.

**Pull request merged**